### ISSUE TITLE
should use detach to make the gradient do not propagate from discriminator to autoencoder 

### ISSUE THREAD

**Issue by tangbohu** [facebookresearch/FaderNetworks#8]
Hi! should we use detach to make the gradient do not propagate from discriminator to autoencoder?

Looking forward to your kind reply!

**Comment by glample** [comment]
Hi,

Where exactly would you want to use `detach`?
If you are training the discriminator, yes it makes sense to use `detach` on the discriminator input to avoid an unnecessary backward through the auto-encoder. This is basically was is done here: https://github.com/facebookresearch/FaderNetworks/blob/master/src/training.py#L97

with `Variable` and `.data` as opposed to `detach`. With the new version of PyTorch, the best now would be to use: `with torch.no_grad():`
